src/orion/algo/dehb/dehb.py

```diff
@@ -0,0 +1,509 @@
+        np.random.seed(seed)
+        self.cs.seed(np.random.randint(np.iinfo(np.int32).max))
+
+    @property
+    def state_dict(self) -> dict:
+        """Return a state dict that can be used to reset the state of the algorithm."""
+        state = dict(self.__dict__)
+        state["client"] = None
+        state["logger"] = None
+        for key in [
+            "active_brackets",
+            "iteration_counter",
+            "de",
+            "_max_pop_size",
+            "start",
+            "traj",
+            "runtime",
+            "history",
+        ]:
+            state[key] = getattr(self, key, None)
+
+        return deepcopy(
+            {
+                "state": state,
+                "numpy_GlobalState": np.random.get_state(),
+                "numpy_RandomState": self.cs.random.get_state(),
+            }
+        )
+
+    def set_state(self, state_dict: dict) -> None:
+        """Reset the state of DEHB"""
+        for k, v in state_dict["state"].items():
+            if hasattr(self, k):
+                setattr(self, k, v)
+            else:
+                logger.error("DEHB does not have attribute %s", k)
+
+        np.random.set_state(state_dict["numpy_GlobalState"])
+        self.cs.random.set_state(state_dict["numpy_RandomState"])
+
+
+# pylint: disable=too-many-public-methods
+class DEHB(DEHBImpl, BaseAlgorithm):
+    """Differential Evolution with HyperBand
+
+    This class is a wrapper around the librairy DEHB:
+    https://github.com/automl/DEHB.
+
+    For more information on the algorithm,
+    see original paper at https://arxiv.org/abs/2105.09821.
+
+    Awad, Noor, Neeratyoy Mallik, and Frank Hutter. "Dehb: Evolutionary hyperband for scalable,
+    robust and efficient hyperparameter optimization." arXiv preprint arXiv:2105.09821 (2021).
+
+    Parameters
+    ----------
+    space: `orion.algo.space.Space`
+        Optimisation space with priors for each dimension.
+
+    seed: None, int or sequence of int
+        Seed for the random number generator used to sample new trials.
+        Default: ``None``
+
+    mutation_factor: float
+        Mutation probability
+        Default: ``0.5``
+
+    crossover_prob: float
+        Crossover probability
+        Default: ``0.5``
+
+    mutation_strategy: str
+        Mutation strategy rand1, rand2dir randtobest1 currenttobest1 best1 best2 rand2
+        Default: ``'rand1'``
+
+    crossover_strategy: str
+        Crossover strategy bin or exp
+        Default: ``'bin'``
+
+    boundary_fix_type: str
+        Boundary fix method, clip or random
+        Default: ``'random'``
+
+    min_clip: float
+        Min clip when boundary fix method is clip
+        Default: ``None``
+
+    max_clip: float
+        Max clip when boundary fix method is clip
+        Default: ``None``
+
+    max_age: Optional[int]
+        Default: ``np.inf``
+
+    """
+
+    requires_type: ClassVar[Optional[str]] = None
+    requires_dist = None
```

Comment:
Can you also add the type variables for the other two class attributes as well?
```suggestion
    requires_dist: ClassVar[Optional[str]] = None
```

Reply:
Added in https://github.com/Epistimio/orion/commit/ebfff61c98c062202355f2ce3c308661e9fc80a1